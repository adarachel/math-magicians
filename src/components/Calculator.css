@import url("https://fonts.googleapis.com/css2?family=Comic+Neue:wght@700&display=swap");

.calculator {
  width: 300px;
  border: 1px solid black;
  border-radius: 5px;
  margin: 0 auto;
  margin-top: 7em;
  padding: 10px;
  background-color: #ececec;
  float: none;
}

.screen {
  height: 50px;
  margin-bottom: 10px;
  padding: 5px;
  background-color: white;
  text-align: right;
  font-size: 24px;
}

.buttons {
  display: flex;
  flex-wrap: wrap;
}

.row {
  display: flex;
  width: 100%;
  margin-bottom: 5px;
}

.button {
  flex: 1;
  padding: 10px;
  margin-right: 5px;
  background-color: #f2f2f2;
  border: none;
  border-radius: 5px;
  font-size: 20px;
  cursor: pointer;
}

.button:hover {
  background-color: #dcdcdc;
}

.operator {
  background-color: #ff8c00;
  color: white;
}

.left-text {
  font-family: "Comic Neue", cursive;
  float: left;
  margin-top: 90px;
  margin-left: 10px;
  font-size: 24px;
}
