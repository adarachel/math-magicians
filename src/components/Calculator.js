import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import Button from './ButtonComponents/Button';

function Calculator() {
  const [display, setDisplay] = useState('0');
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = (buttonName) => {
    if (buttonName === 'AC') {
      setTotal(null);
      setNext(null);
      setOperation(null);
      setDisplay('0');
      return;
    }
    const result = calculate({ total, next, operation }, buttonName);
    if (result) {
      setTotal(result.total);
      setNext(result.next);
      setOperation(result.operation);
      setDisplay((display === '0' && buttonName !== '.') ? buttonName : display + buttonName);
    }
    if (buttonName === '=') {
      setDisplay(result.total);
      setTotal(result.total);
      setNext(null);
      setOperation(null);
    }
  };

  return (
    <div className="calculator">
      <div className="screen">{display}</div>
      <div className="buttons">
        <div className="row">
          <Button className="button" name="AC" handleClick={() => handleClick('AC')} />
          <Button className="button" name="+/-" handleClick={() => handleClick('+/-')} />
          <Button className="button" name="%" handleClick={() => handleClick('%')} />
          <Button className="button operator" name="รท" handleClick={() => handleClick('รท')} />
        </div>
        <div className="row">
          <Button className="button" name="7" handleClick={() => handleClick('7')} />
          <Button className="button" name="8" handleClick={() => handleClick('8')} />
          <Button className="button" name="9" handleClick={() => handleClick('9')} />
          <Button className="button operator" name="x" handleClick={() => handleClick('x')} />
        </div>
        <div className="row">
          <Button className="button" name="4" handleClick={() => handleClick('4')} />
          <Button className="button" name="5" handleClick={() => handleClick('5')} />
          <Button className="button" name="6" handleClick={() => handleClick('6')} />
          <Button className="button operator" name="-" handleClick={() => handleClick('-')} />
        </div>
        <div className="row">
          <Button className="button" name="1" handleClick={() => handleClick('1')} />
          <Button className="button" name="2" handleClick={() => handleClick('2')} />
          <Button className="button" name="3" handleClick={() => handleClick('3')} />
          <Button className="button operator" name="+" handleClick={() => handleClick('+')} />
        </div>
        <div className="row">
          <Button className="button" name="0" handleClick={() => handleClick('0')} />
          <Button className="button" name="." handleClick={() => handleClick('.')} />
          <Button className="button operator" name="=" handleClick={() => handleClick('=')} />
        </div>
      </div>
    </div>
  );
}

export default Calculator;
