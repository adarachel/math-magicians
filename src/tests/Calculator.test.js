import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

test('Calculator component renders correctly', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchSnapshot();
});

test('Calculator component displays initial value of 0', () => {
  const { getByText } = render(<Calculator />);
  const displayElement = getByText('0');
  expect(displayElement).toBeInTheDocument();
});

test('Calculator component updates display with clicked button values', () => {
  const { getByText } = render(<Calculator />);
  const buttonElement = getByText('1');
  fireEvent.click(buttonElement);
  const displayElement = getByText('1');
  expect(displayElement).toBeInTheDocument();
});

test('Calculator component performs addition correctly', () => {
  const { getByText } = render(<Calculator />);
  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('='));
  const displayElement = getByText('3');
  expect(displayElement).toBeInTheDocument();
});
