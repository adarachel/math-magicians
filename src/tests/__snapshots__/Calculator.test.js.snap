// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="left-text"
    >
      Let's do some math!
    </div>
    <div
      class="calculator"
    >
      <div
        class="screen"
      >
        0
      </div>
      <div
        class="buttons"
      >
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            AC
          </button>
          <button
            class="button"
            type="button"
          >
            +/-
          </button>
          <button
            class="button"
            type="button"
          >
            %
          </button>
          <button
            class="button operator"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            7
          </button>
          <button
            class="button"
            type="button"
          >
            8
          </button>
          <button
            class="button"
            type="button"
          >
            9
          </button>
          <button
            class="button operator"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            4
          </button>
          <button
            class="button"
            type="button"
          >
            5
          </button>
          <button
            class="button"
            type="button"
          >
            6
          </button>
          <button
            class="button operator"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            1
          </button>
          <button
            class="button"
            type="button"
          >
            2
          </button>
          <button
            class="button"
            type="button"
          >
            3
          </button>
          <button
            class="button operator"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            0
          </button>
          <button
            class="button"
            type="button"
          >
            .
          </button>
          <button
            class="button operator"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Calculator page component renders correctly 1`] = `
<div>
  <div
    className="left-text"
  >
    Let's do some math!
  </div>
  <div
    className="calculator"
  >
    <div
      className="screen"
    >
      0
    </div>
    <div
      className="buttons"
    >
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Calculator renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="left-text"
    >
      Let's do some math!
    </div>
    <div
      class="calculator"
    >
      <div
        class="screen"
      >
        0
      </div>
      <div
        class="buttons"
      >
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            AC
          </button>
          <button
            class="button"
            type="button"
          >
            +/-
          </button>
          <button
            class="button"
            type="button"
          >
            %
          </button>
          <button
            class="button operator"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            7
          </button>
          <button
            class="button"
            type="button"
          >
            8
          </button>
          <button
            class="button"
            type="button"
          >
            9
          </button>
          <button
            class="button operator"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            4
          </button>
          <button
            class="button"
            type="button"
          >
            5
          </button>
          <button
            class="button"
            type="button"
          >
            6
          </button>
          <button
            class="button operator"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            1
          </button>
          <button
            class="button"
            type="button"
          >
            2
          </button>
          <button
            class="button"
            type="button"
          >
            3
          </button>
          <button
            class="button operator"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            0
          </button>
          <button
            class="button"
            type="button"
          >
            .
          </button>
          <button
            class="button operator"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Calculator component correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="left-text"
    >
      Let's do some math!
    </div>
    <div
      class="calculator"
    >
      <div
        class="screen"
      >
        0
      </div>
      <div
        class="buttons"
      >
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            AC
          </button>
          <button
            class="button"
            type="button"
          >
            +/-
          </button>
          <button
            class="button"
            type="button"
          >
            %
          </button>
          <button
            class="button operator"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            7
          </button>
          <button
            class="button"
            type="button"
          >
            8
          </button>
          <button
            class="button"
            type="button"
          >
            9
          </button>
          <button
            class="button operator"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            4
          </button>
          <button
            class="button"
            type="button"
          >
            5
          </button>
          <button
            class="button"
            type="button"
          >
            6
          </button>
          <button
            class="button operator"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            1
          </button>
          <button
            class="button"
            type="button"
          >
            2
          </button>
          <button
            class="button"
            type="button"
          >
            3
          </button>
          <button
            class="button operator"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="button"
            type="button"
          >
            0
          </button>
          <button
            class="button"
            type="button"
          >
            .
          </button>
          <button
            class="button operator"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly when there are no items 1`] = `
<div>
  <div
    className="left-text"
  >
    Let's do some math!
  </div>
  <div
    className="calculator"
  >
    <div
      className="screen"
    >
      0
    </div>
    <div
      className="buttons"
    >
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button operator"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
